third_party/ienn/src/ie_execution.cpp

```diff
@@ -173,17 +173,21 @@ int32_t Execution::StartCompute() {
       OutputsDataMap outputInfo = compilation_->network_->getOutputsInfo();
       auto outputInfoItem = outputInfo.begin();
       Blob::Ptr output_blob = infer_request->GetBlob(outputInfoItem->first);
-      outputInfoItem ++;
       if (operand.dimensions.size() == 3) {
         const float* src =
             output_blob->buffer()
                 .as<PrecisionTrait<Precision::FP32>::value_type*>();
+        std::vector<unsigned int> dimensions(operand.dimensions);
+        if (outputInfoItem->first.find("TopK") != outputInfoItem->first.npos) {
+          // Reshape 3D to 4D
+          dimensions.push_back(1);
```

Comment:
It's only used for NHWC?

Reply:
The layout nnapi support currently is only NHWC?

Reply:
Could you please add some explainer why do we need this special case?

Reply:
Yes, I will add more comments to explain.